src/misc/fs_utils_posix.cpp

```diff
@@ -329,8 +337,16 @@ uint16_t local_drive_set_attributes(const std_fs::path& path,
 
 	bool status = make_writable(path);
 	if (status) {
+		struct stat file_info;
+		bool is_directory = true;
+		if (stat(path.c_str(), &file_info) == 0) {
+			is_directory = file_info.st_mode & S_IFDIR;
+		}
 		status = set_xattr(path, attributes);
-		if (status && attributes.read_only) {
+		if (status && attributes.read_only && !is_directory) {
```

Comment:
Why are we skipping this for directories? The local_drive_get_attributes checks the filesystem attributes for read only flag, too.